Added ASF-required links using drop-down menu and unified navigation

zka26 opened the pull request:
<!-- Thank you for submitting a pull request to our repo. -->

<!-- Please do NOT submit PRs for features in newer versions of Lucene. We should keep the target version consistent across our repository to make the upgrade process to newer versions of Lucene go smoothly. -->

<!-- If this is your first PR in the Lucene.NET repo, please run through the checklist
below to ensure a smooth review and merge process for your PR. -->

- [x] You've read the [Contributor Guide](https://github.com/apache/lucenenet/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://www.apache.org/foundation/policies/conduct.html).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a change, please [open an issue](https://github.com/apache/lucenenet/issues/new/choose) to discuss the change or find an existing issue.

<!-- Once all that is done, you're ready to go. Open the PR with the content below. -->

Added ASF-required links via dropdown and unified navigation.

Fixes #1186 

## Description

This change attempts to standardize the navbar with the newly introduced ASF menus across the Site and Apidocs and, to fix any sort of layout glitches and to improve zoom and resize and to address and solve problem that could have arisen from what was changed. DevTools were used to ensure that everything works accordingly.


### General Design:

- Mobile < 768
- ~769 < Compact < 1199~
- **(New)** 768 =< Wrap < 1200
- 1200 <= Desktop

**Desktop**: Classic horizontal navigation bar extended with the ASF drop-down menu upon hover. The header is fixed when scrolling.

**Mobile:** Center-aligned column menu with a tap/click-to-open ASF accordion.

~**Compact**: Transition in-between Mobile and Desktop for legacy resolutions (or maybe some tablets). It uses the hamburger menu already and the header is no longer fixed when scrolling focusing on the visibility of the content while the ‘Table of Content’ still appears as the left sidebar, and to a certain point the ‘In this Article’ appears as the right sidebar.~

**(New) Wrap:** Reintroduced that the navbar items ( + searchbar) wraps under the logo.

### Improvements:

- Unified navigation: Same feel and look across the Site and Apidocs.
- ~Compact mode (with new breakpoints): When the single classic horizontal navigation bar is no longer possible, we turn to the hamburger menu to prevent the header covering up too much of the content.~
**(New)** Wrap mode instead of the compact mode. The buttons' heights were decreased slightly.
- Ribbon safety: Keeps the hamburger menu clear of the “Fork me on GitHub” ribbon, so it is always accessible, never covered by it.
- [Apidocs-Only] Search overlay: In-between 1400px and 1200px instead of the base search function, we temporarily introduce a search button to open an overlay, therefore delaying the need to switch to hamburger (or multi-line) too early, and Apidocs can switch to it at the same time as the Site.

	
### Previews
#### Site:
<details>
  <summary> Site - Desktop  </summary>
<img width="1305" height="823" alt="image" src="https://github.com/user-attachments/assets/039c282e-ab22-4948-9347-fbbc0fefaab2" />
 </details>
 
 <details>
  <summary> Site - Wrap </summary>
<img width="812" height="827" alt="image" src="https://github.com/user-attachments/assets/c645dec3-75ab-4f05-aebc-ea5a5a871f16" />

 </details>
 
 <details>
  <summary> Site - Mobile </summary>
<img width="530" height="831" alt="image" src="https://github.com/user-attachments/assets/61c29c00-d489-4327-b068-3c7ece21ad5b" />

 </details>

 
 
 #### Apidocs:
 <details>
  <summary> Apidocs - Desktop: </summary>
<img width="1283" height="833" alt="image" src="https://github.com/user-attachments/assets/9f9309d1-733e-4690-a3c3-65a7a504df7d" />
 </details>
 
  <details>
  <summary> Apidocs - Wrap: </summary>
<img width="882" height="827" alt="image" src="https://github.com/user-attachments/assets/8ffc34cc-52c9-4257-a222-ed64be65c55a" />

 </details>
 
  <details>
  <summary> Apidocs - Mobile: </summary>
<img width="555" height="836" alt="image" src="https://github.com/user-attachments/assets/6bb0bd0a-3b45-4332-aff9-1fd1155f5005" />
 </details>

paulirwin commented:
I do like the dropdown in the header, but I'm also okay with the footer for now. 

The header is already wrapping on smaller resolutions, like the tablet I'm using now. If we do the header change, maybe we could move Contributing under Documentation, and Download under About (or remove Download, if possible, since most people are probably using NuGet). 

But, I know this is a Hacktoberfest item, so I'm also cool with taking the quickest path to victory here.

NightOwl888 commented:
> remove Download, if possible, since most people are probably using NuGet

An Apache release is *source code*, so these pages are required to download the official releases. The pre-built binaries and packages are technically only provided on NuGet for convenience (at least in Apache's eyes).

https://infra.apache.org/release-publishing.html#valid

That said, IIRC there is an official `.cgi` file that we could style that would dynamically show the releases on Apache's archives.

zka26 commented:
Thanks a lot for the detailed feedback and the examples! I would like to work on it and do it the right way, using drop-down approach in the header and try adjusting the menu layout while testing on smaller screen. I will work on that and update the PR accordingly and at the end we can see, if anything else needs to be done, more than glad to contribute! Thank you for the guidance again!

NightOwl888 commented:
@zka26 - I noticed this is still marked as a draft PR. Do you have more work to do or is this ready for review?

Note that the EditorConfig checker also caught some issues with extra trailing whitespace.

zka26 commented:
Sorry, this took longer than expected. The learning curve was trickier than I thought and I decided to restart a few times. I added the ASF-required links via dropdown and unified the navs. I significantly reworked the main.css and main.js files (added main.js for the Site). I tested across viewports/zoom in DevTools to address issues. On the Apidocs, there is a compact search overlay between 1200-1399px to keep a single-line navbar and match the Site. I hope the result feels clean and consistent.
If anything should be reverted, adjusted, included or excluded, please let me know!
I am happy to improve it and take your recommendation.

Edit: Please take all the time you need on the review. Hacktoberfest isn’t a deadline for me.

NightOwl888 reviewed (approved):
Thanks. This looks really good.

NightOwl888 commented:
FYI - these changes are now live. Thanks again for the contribution.

After I deployed the site, I realized there were 2 barely noticeable style issues on the API docs site.

1. The ASF dropdown menu only appears on the [home page of the docs site](https://lucenenet.apache.org/), not any of the other pages, like [Lucene.Net.Analysis.Common](https://lucenenet.apache.org/docs/4.8.0-beta00017/api/analysis-common/overview.html).
2. The breadcrumb menu used to begin with "API /" and then every breadcrumb would follow. But now the "API" part is on its own line and the rest of the breadcrumb appears below it. To see what I mean, compare [ArabicNormalizationFilterFactory on 4.8.0-beta00017](https://lucenenet.apache.org/docs/4.8.0-beta00017/api/analysis-common/Lucene.Net.Analysis.Ar.ArabicNormalizationFilterFactory.html) with [ArabicNormalizationFilterFactory on 4.8.0-beta00016](https://lucenenet.apache.org/docs/4.8.0-beta00016/api/analysis-common/Lucene.Net.Analysis.Ar.ArabicNormalizationFilterFactory.html).

Feel free to submit another PR to fix these, but as far as Apache is concerned we have fixed the ASF menu issue because it was only required on the home page of the main site.

The right menu is also missing from the 4.8.0-beta00017 docs (the part that indexes the current page), but I think that may be a docfx config issue instead of a style issue.

paulirwin commented:
Wow, that was aggressively cached. I had done that but it didn't work the first time. Great to see this is live!

However, the Whimsy checks are still failing as of this morning. We'll need to dig in to figure out why, but as long as we actually have the links there, at least we're meeting the requirement. Or, perhaps they are having a caching issue too that will eventually clear. https://whimsy.apache.org/site/project/lucenenet
